test-zuul/src/main/resources/application.yml

```diff
@@ -2,12 +2,19 @@ spring:
   application:
     name: ZUUL
 server:
+  address: 127.0.0.1
   port: 8765
 eureka:
   instance:
     metadataMap:
       instanceId: ${spring.application.name}-${random.number}
+    ip-address: 127.0.0.1
+    hostname: localhost
+    prefer-ip-address: false
   client:
     enabled: true
     serviceUrl:
-      defaultZone: http://${eureka.host:localhost}:${eureka.port:1111}/eureka/
\ No newline at end of file
+      defaultZone: http://${eureka.host:localhost}:${eureka.port:1111}/eureka/
+    use-dns-for-fetching-service-urls: false
+zuul:
+  sensitive-headers: Authorization
```

Comment:
`zuul. sensitive-headers` controls what HTTP headers are blocked by Zuul. By default `Set-Cookie` and `Cookie` headers are blocked. This causes problems for an interactive Keycloak login.

First, Keycloak's `OAuth_Token_Request_State` cookie cannot be saved. This causes Keycloak to be unable to verify the authenticity of the login request in the code-to-token phase.

Second, since the application cannot set a `JSESSION_ID` cookie, the original request URL is lost and the result of the code-to-token phase cannot complete the authentication and save it to the Spring Security context.